src/v/cloud_storage/segment_chunk_api.cc

```diff
@@ -149,10 +150,64 @@ ss::future<segment_chunk::handle_t> segment_chunks::hydrate_chunk(
       chunk_start,
       chunk.waiters.size());
 
+    auto prefetch = prefetch_override.value_or(
+      config::shard_local_cfg().cloud_storage_chunk_prefetch);
+    schedule_prefetches(chunk_start, prefetch);
+
     _bg_cvar.signal();
     co_return co_await std::move(f);
 }
 
+void segment_chunks::schedule_prefetches(
+  chunk_start_offset_t start_offset, size_t prefetch) {
+    for (auto it = std::next(_chunks.find(start_offset));
```

Comment:
Should we vassert that `_chunks.find() != _chunks.end()`? I guess we do this above but in isolation this reads like a bug